client/lib/media/store.js

```diff
@@ -123,6 +123,12 @@ MediaStore.dispatchToken = Dispatcher.register( function( payload ) {
 
 			MediaStore.emit( 'change' );
 			break;
+		case 'FETCH_MEDIA_LIMITS':
+			if ( ! action.siteId ) {
+				break;
+			}
+			MediaStore.emit( 'fetch-media-limits' );
```

Comment:
You've committed a sin by emitting an event other than `change` here :smile: But at this point, those sorts of revisions should further motivate us to porting this to Redux. An alternative approach would be to create an entirely new Flux store simply responsible for emitting a `change` event in response to `FETCH_MEDIA_LIMITS` (which probably isn't as much as work as it might sound).

Now that I think of it, do we need a custom action type for `FETCH_MEDIA_LIMITS`. Isn't it enough to hook `RECEIVE_MEDIA_ITEM` and `REMOVE_MEDIA_ITEM` server actions (i.e. `payload.source === 'SERVER_ACTION'`)?